packages/xo-web/src/xo-app/menu/index.js

```diff
@@ -25,12 +26,62 @@ import {
   getXoaState,
   isAdmin,
 } from 'selectors'
-import { every, identity, isEmpty, map } from 'lodash'
+import { every, forEach, identity, isEmpty, map } from 'lodash'
 
 import styles from './index.css'
 
 const returnTrue = () => true
 
+@connectStore(() => ({
+  hosts: createGetObjectsOfType('host'),
+}))
+class MissingPatchWarning extends Component {
+  state: { haveMissingPatches: false }
+
+  componentWillMount() {
+    this._subscribeMissingPatches()
+  }
+
+  componentWillUnmount() {
+    this.unsubscribeMissingPatches()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.hosts !== this.props.hosts) {
```

Comment:
`hosts` is an array ?

Reply:
If it is an array, use Array.proptypes.equals instead.

Reply:
No, it's an object.

Reply:
I see .. here you're comparing their references. If you want to compare their values, use `lodash.isEqual` instead.